File: 744 - Find Smallest Letter Greater Than Target.java
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        int l = 0, r = letters.length - 1;
        int m;
        while (l <= r) {
            m = (l + r) / 2;
            if (letters[m] > target) {
                r = m - 1;
            } else {
                l = m + 1;
            }
        }
        if (l == letters.length) {
            return letters[0];
        } else {
            return letters[l];
        }

    }
}

File: 744 - Find Smallest Letter Greater Than Target.py
class Solution:
    def nextGreatestLetter(self, letters: list[str], target: str) -> str:
        i = self.find(letters, target)
        return i
    def find(self, letters, target):
        l = 0
        r = len(letters) - 1
        # found = False
        while l <= r:
            m = (l + r) // 2
            if letters[m] > target:
                r = m - 1
            else:
                l = m + 1
        if l == len(letters):
            return letters[0]
        else:
            return letters[l]

